plugin.video.xumotv/resources/lib/xumotv.py

```diff
@@ -0,0 +1,374 @@
+            playlist.add(url, liz, idx)
+            if idx == 0: xbmcplugin.setResolvedUrl(int(sys.argv[1]), True, liz)
+
+            
+    def uEPG(self):
+        log('uEPG')
+        #support for uEPG universal epg framework module. https://github.com/Lunatixz/KODI_Addons/tree/master/script.module.uepg
+        stations = sorted(list(self.getChannels()), key=lambda item: item['number'])
+        listings = sorted(self.getOnNEXT(), key=lambda item: item['start'], reverse=False)
+        return [self.buildGuide((station, listings)) for station in stations]
+        
+        
+    def buildGuide(self, item):
+        log('buildGuide')
+        station, listings = tuple(item)
+        chname     = station['title']
+        chnum      = station['number']
+        chid       = station['guid']['value']
+        genre      = station['genre'][0]['value']
+        chlogo     = BASE_LOGO%(chid)
+        newChannel = {}
+        guidedata  = []
+        newChannel['channelname']   = chname
+        newChannel['channelnumber'] = chnum
+        newChannel['channellogo']   = chlogo
+        newChannel['isfavorite']    = False
+        endTime    = datetime.datetime.now()
+        for listing in listings:
+            if listing['channelId'] != int(chid): continue
+            cid   = listing['id']
+            meta  = self.getMeta(cid)
+            label       = meta['title']
+            plot        = (self.getDescription(meta.get('descriptions',{})) or label)
+            try: aired  = datetime.datetime.strptime(meta['availableSince'],'%a, %d %b %Y %H:%M:%S %z')
+            except: aired = datetime.datetime.now()
+            duration    = meta.get('runtime',5200)
+            startTime   = endTime
+            endTime     = startTime + datetime.timedelta(seconds=duration)
+            try: url    = meta['providers'][0]['sources'][0]['uri']
+            except: url = chid
+            tmpdata = {"mediatype":"episode","label":label ,"title":label,"plot":plot,"duration":duration,"aired":aired.strftime('%Y-%m-%d'),'genre':genre}
+            tmpdata['starttime'] = int(time.mktime((startTime).timetuple()))
+            tmpdata['url'] = self.sysARG[0]+'?mode=9&name=%s&url=%s'%(urllib.quote_plus(uni(chname)),url)
+            tmpdata['art'] = {'thumb':BASE_THUMB%(cid),'fanart':BASE_FANART%(chid),'clearlogo':BASE_LOGO%(chid)}
+            guidedata.append(tmpdata)
+        newChannel['guidedata'] = guidedata
+        return newChannel
+
+        
+    def addLink(self, name, u, mode, liz, total=0):
+        name = name.encode("utf-8")
+        log('addLink, name = ' + name)
+        u=self.sysARG[0]+"?url="+urllib.quote_plus(u)+"&mode="+str(mode)+"&name="+urllib.quote_plus(name)
+        xbmcplugin.addDirectoryItem(handle=int(self.sysARG[1]),url=u,listitem=liz,totalItems=total)
+
+
+    def addDir(self, name, u, mode, infoList=False, infoArt=False):
+        name = name.encode("utf-8")
+        log('addDir, name = ' + name)
```

Comment:
unnecessary log spam

Reply:
@enen92 No log spamming, all logging is voluntary via enabling debugging in settings.
https://github.com/Lunatixz/repo-plugins/blob/9ee6be28c55af148764b21dcd2b609460f4f85e2/plugin.video.xumotv/resources/lib/xumotv.py#L54
https://github.com/Lunatixz/repo-plugins/blob/9ee6be28c55af148764b21dcd2b609460f4f85e2/plugin.video.xumotv/resources/lib/xumotv.py#L41